editor/src/clj/editor/sprite.clj

```diff
@@ -453,91 +496,111 @@
+      3 types/Vec3
+      4 types/Vec4)))
+
+(defn- attribute-expected-element-count [attribute]
+  (case (:semantic-type attribute)
+    :semantic-type-color 4
+    (:element-count attribute)))
 
 (defn- attribute-update-property [current-property-value attribute new-value]
   (assoc current-property-value (:name-key attribute) new-value))
 
 (defn- attribute-clear-property [current-property-value attribute]
   (dissoc current-property-value (:name-key attribute)))
 
-(defn- get-attribute-edit-type [attribute prop-type]
-  (let [attribute-semantic-type (:semantic-type attribute)
+(defn- attribute-edit-type [attribute property-type]
+  (let [attribute-element-count (:element-count attribute)
+        attribute-semantic-type (:semantic-type attribute)
         attribute-update-fn (fn [_evaluation-context self _old-value new-value]
-                              (g/update-property self :vertex-attribute-overrides attribute-update-property attribute new-value))
+                              (let [values (if (= g/Num property-type)
+                                             (vector-of :double new-value)
+                                             new-value)]
+                                (g/update-property self :vertex-attribute-overrides attribute-update-property attribute values)))
         attribute-clear-fn (fn [self _property-label]
-                              (g/update-property self :vertex-attribute-overrides attribute-clear-property attribute))]
-    (if (= attribute-semantic-type :semantic-type-color)
-      {:type types/Color
-       :ignore-alpha? false
-       :set-fn attribute-update-fn
-       :clear-fn attribute-clear-fn}
-      {:type prop-type
-       :set-fn attribute-update-fn
-       :clear-fn attribute-clear-fn})))
-
-(defn- attributes->intermediate-backing [attributes]
-  (into {}
-        (map (fn [attribute]
-               [(vtx/attribute-name->key (:name attribute))
-                (graphics/attribute->any-values attribute)]))
-        attributes))
+                             (g/update-property self :vertex-attribute-overrides attribute-clear-property attribute))]
+    (cond-> {:type property-type
+             :set-fn attribute-update-fn
+             :clear-fn attribute-clear-fn}
+
+            (= attribute-semantic-type :semantic-type-color)
+            (assoc :ignore-alpha? (not= 4 attribute-element-count)))))
+
+(defn- attribute-key->property-key-raw [attribute-key]
+  (keyword (str "attribute_" (name attribute-key))))
+
+(def ^:private attribute-key->property-key (memoize attribute-key->property-key-raw))
+
+(def ^:private attribute-info->property-key (comp attribute-key->property-key :name-key))
 
-(defn- attribute->outline-key [attribute-info]
-  (keyword (str "attribute_" (name (:name-key attribute-info)))))
+(defn- editable-attribute-info? [attribute-info]
+  (case (:semantic-type attribute-info)
+    (:semantic-type-position :semantic-type-texcoord :semantic-type-page-index) false
+    true))
```

Comment:
* Hide any material-declared `position`, `texcoord`, and `page_index` attributes of sprites in the Property Editor.